src/manifests_workflow/component_opensearch_min.py

```diff
@@ -23,19 +26,31 @@ def __init__(self, repo: GitRepository, snapshot: bool = False) -> None:
         )
 
     @classmethod
-    def branches(self, url: str = "https://github.com/opensearch-project/OpenSearch.git") -> List[str]:
+    def branches(cls, url: str = "https://github.com/opensearch-project/OpenSearch.git") -> List[str]:
         return Component.branches(url)
 
     @classmethod
-    def checkout(self, path: str, branch: str = "main", snapshot: bool = False) -> 'ComponentOpenSearchMin':
+    def checkout(cls, path: str, branch: str = "main", snapshot: bool = False) -> 'ComponentOpenSearchMin':
+        cls.path = path
         return ComponentOpenSearchMin(
             GitRepository("https://github.com/opensearch-project/OpenSearch.git", branch, path),
             snapshot,
         )
 
     @property
     def properties(self) -> PropertiesFile:
-        cmd = ComponentOpenSearch.gradle_cmd("properties", {"build.snapshot": str(self.snapshot).lower()})
+        min_comp_version_path = os.path.join(self.path, "buildSrc", "src", "main", "resources", "minimumCompilerVersion")
+        # Trying to read the minimumCompilerVersion file
+        # And force gradle to apply java home path defined with env var JAVA<Version>_HOME, i.e. JAVA11_HOME
+        # If file is not found then fallback to the default java home defined by host
+        java_home_path = None
+        with open(min_comp_version_path, "r") as file:
+            java_home_path = os.getenv(f"JAVA{file.read().strip()}_HOME", None)
```

Comment:
Do we want to put this under try/catch block?

Reply:
We dont, if there is no env found, it will just proceed with original command, and use default java_home or java env set by the host or alternatives.

Reply:
Wondering in case that file is missing it will just error out or python is unable to open in read mode.

Reply:
That is fine, if the file is missing it will error out, and we will update the core repo.
If the file is there but version is old, aka 14 that we dont have installed on docker, then it will just use the default without specifying the jdk version.